python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -36,6 +41,160 @@ class SGDBackendError(Exception):
     """Errors with BackendExecutor that should not be exposed to user."""
 
 
+class CheckpointManager:
+    """Manages checkpoint processing, writing, and loading.
+
+    Directory structure:
+    - A logdir is created during instantiation. This will hold all the
+    results/checkpoints for the lifetime of the Trainer. By default, it will be
+    of the form ``~/ray_results/sgd_<datestring>``.
+    - A run_dir is created every time ``start_training`` is called. This will
+    hold the checkpoints and results for a single ``trainer.run()`` or
+    ``trainer.run_iterator()`` call. It will be of the form ``run_<run_id>``.
+    - A ``checkpoints`` directory is created in the ``run_dir`` and contains
+    all the checkpoint files.
+
+    The full default path will be:
+
+    ~/ray_results/sgd_<datestring>/run_<run_id>/checkpoints/
+    checkpoint_<checkpoint_id>
```

Comment:
Logdir/rundir docs can be moved out to `Trainer` - at this point we only care about `run_dir` as the parent directory.

Reply:
Moved!